internal/cmd/keys.go

```diff
@@ -68,35 +65,23 @@ infra keys add first-key bot --ttl=12h --extension-deadline=1h
 				return err
 			}
 
-			deadline := ThirtyDays
-			if options.ExtensionDeadline != "" {
-				deadline, err = time.ParseDuration(options.ExtensionDeadline)
-				if err != nil {
-					return err
-				}
-			}
-
-			ttl := ThirtyDays
-			if options.TTL != "" {
-				ttl, err = time.ParseDuration(options.TTL)
-				if err != nil {
-					return fmt.Errorf("parsing ttl: %w", err)
-				}
-			}
-
-			resp, err := client.CreateAccessKey(&api.CreateAccessKeyRequest{IdentityID: machine.ID, Name: keyName, TTL: api.Duration(ttl), ExtensionDeadline: api.Duration(deadline)})
+			resp, err := client.CreateAccessKey(&api.CreateAccessKeyRequest{
+				IdentityID:        machine.ID,
+				Name:              keyName,
+				TTL:               api.Duration(options.TTL),
+				ExtensionDeadline: api.Duration(options.ExtensionDeadline),
+			})
 			if err != nil {
 				return err
 			}
 
 			fmt.Printf("key: %s \n", resp.AccessKey)
-
 			return nil
 		},
 	}
 
-	cmd.Flags().String("ttl", "", "The total time that an access key will be valid for, defaults to 30 days")
-	cmd.Flags().String("extension-deadline", "", "A specified deadline that an access key must be used within to remain valid, defaults to 30 days")
+	cmd.Flags().DurationVar(&options.TTL, "ttl", ThirtyDays, "The total time that an access key will be valid for, defaults to 30 days")
+	cmd.Flags().DurationVar(&options.ExtensionDeadline, "extension-deadline", ThirtyDays, "A specified deadline that an access key must be used within to remain valid, defaults to 30 days")
```

Comment:
This change also improves the help text of the command, because now the default value (30 days) is visible from the help  text.